code/modules/crafting/slapcrafting/_crafting_stage.dm

```diff
@@ -76,17 +76,18 @@
 	. = !consume_completion_trigger || user.try_unequip(thing, target)
 	if(. && stack_consume_amount > 0)
 		var/obj/item/stack/stack = thing
-		if(!istype(stack) || stack.amount < stack_consume_amount)
+		if(!istype(stack) || stack.get_amount() < stack_consume_amount)
 			on_insufficient_material(user, stack)
 			return FALSE
 		var/obj/item/stack/used_stack
-		if(stack.amount == stack_consume_amount)
+		if(stack.amount > stack_consume_amount)
+			used_stack = stack.split(stack_consume_amount)
+		else
 			if(!user.try_unequip(thing, target))
 				return FALSE
 			used_stack = stack
-		else
-			used_stack = stack.split(stack_consume_amount)
-		used_stack.forceMove(target)
+		if(used_stack)
```

Comment:
```suggestion
		if(!QDELETED(used_stack))
```
unsure if this is necessary or not but it probably wouldn't hurt if you feel up to changing it